processing/db_handler.py

```diff
@@ -233,44 +235,52 @@ def get_new_data_upload_log(self,
                          'ON u.log_id = x.upload_log_id '
                          'LEFT JOIN input_interface.data_source_type s '
                          'ON u.upload_source_id = s.source_id '
+                         'LEFT JOIN '
+                         'input_interface.data_upload_type t '
+                         'ON u.upload_type_id = t.type_id '
                          'WHERE p.log_id IS NULL '
-                         'AND u.upload_type_id IN (4, 7) '
+                         'AND t.description IN (\'Half hourly data\', '
+                         '\'Half-hourly gap-filled data\') '
                          'AND x.xfer_end_log_timestamp IS NOT NULL ')
+        composed_query_list = [query_base]
         if is_qaqc_processor:
-            query_add_1 = SQL('AND s.source = %(qaqc_processor_source)s')
+            composed_query_list.append(
+                SQL('AND s.source = %(qaqc_processor_source)s'))
         else:
-            query_add_1 = SQL('AND s.source != %(qaqc_processor_source)s')
-        query = Composed([query_base, query_add_1])
+            composed_query_list.append(
+                SQL('AND s.source <> %(qaqc_processor_source)s'))
         params = {'qaqc_processor_source': qaqc_processor_source}
         if uuid:
-            query_add_2 = SQL('AND u.upload_token = %(uuid)s')
-            query = Composed([query, query_add_2])
+            composed_query_list.append(
+                SQL('AND u.upload_token = %(uuid)s'))
             params['uuid'] = uuid
+        query = Composed(composed_query_list)
         with conn.cursor(cursor_factory=RealDictCursor) as cursor:
             cursor.execute(query, params)
             new_data_upload = cursor.fetchall()
         return new_data_upload
 
     def get_latest_run_with_uuid(self, conn, uuid):
-        query = SQL('SELECT u.log_id, p.process_timestamp '
-                    'FROM input_interface.data_upload_log u '
-                    'LEFT JOIN qaqc.processing_log p '
-                    'ON u.log_id = p.upload_id '
-                    'WHERE u.upload_token = %(uuid)s '
-                    'ORDER BY p.process_timestamp DESC '
-                    'LIMIT 1')
+        query = SQL('SELECT u.log_id, latest_upload.process_timestamp FROM '
+                    'input_interface.data_upload_log u '
+                    'INNER JOIN (SELECT upload_id, '
+                    'max(process_timestamp) AS process_timestamp '
+                    'FROM qaqc.processing_log p GROUP BY log_id) '
```

Comment:
Probably a good idea to have "FROM" be at the start of the next line, and just have consistent SQL formatting.